src/main/java/org/truffleruby/core/queue/SizedQueueNodes.java

```diff
@@ -134,8 +121,37 @@ private void doPushBlocking(final Object value, final SizedQueue queue) {
             });
         }
 
+        @Specialization(guards = "!nonBlocking")
+        protected Object pushBlocking(
+                RubySizedQueue self, final Object value, boolean nonBlocking, long timeoutMilliseconds) {
+            final SizedQueue queue = self.queue;
+            propagateSharingNode.executePropagate(self, value);
+            final long deadline = System.currentTimeMillis() + timeoutMilliseconds;
+
+            var final_result = getContext().getThreadManager().runUntilResult(this, () -> {
```

Comment:
```suggestion
            var finalResult = getContext().getThreadManager().runUntilResult(this, () -> {
```
Or clearer: `boolean success`

Reply:
Updated.